dev/NavigationView/NavigationView.cpp

```diff
@@ -1358,6 +1362,36 @@ void NavigationView::UpdateIsClosedCompact()
     }
 }
 
+void NavigationView::UpdatePaneButtonsWidths()
+{
+    auto newButtonWidths = [this]()
+    {
+        if (DisplayMode() == winrt::NavigationViewDisplayMode::Minimal)
+        {
+            return static_cast<double>(c_paneToggleButtonWidth);
+        }
+        return CompactPaneLength();
+    }();
+
+    if (auto&& backButton = m_backButton.get())
+    {
+        backButton.Width(newButtonWidths);
+    }
+    if (auto&& paneToggleButton = m_paneToggleButton.get())
+    {
+        paneToggleButton.MinWidth(newButtonWidths);
+        if (const auto iconGridColumnElement = paneToggleButton.GetTemplateChild(c_paneToggleButtonIconGridColumnName))
+        {
+            if (const auto paneToggleButtonIconColumn = iconGridColumnElement.try_as<winrt::ColumnDefinition>())
+            {
+                auto width = paneToggleButtonIconColumn.Width();
+                width.Value = newButtonWidths;
+                paneToggleButtonIconColumn.Width(width);
```

Comment:
Yes that should work, but it's propably faster if we handle that switching ourselves.

Reply:
The binding approach would be more robust though. For example, if the pane toggle button is initially hidden (`IsPaneToggleButtonVisible="False"`) and later made visible, then this piece of code won't be able to get the icon column since the button's template won't be applied until it's visible. To handle this (rare) situation, additional logic would need to be added, which could result in more complexity than necessary.

Reply:
Since we don't lazy load the control, as soon as we "ask for it" through `GetTemplateChild` the element get's realized and we can change it's state, so this should be a non issue right now.